clarity/src/vm/functions/database.rs

```diff
@@ -924,3 +947,190 @@ pub fn special_get_burn_block_info(
+                .database
+                .get_block_header_hash(height_value)?;
+            Value::Sequence(SequenceData::Buffer(BuffData {
+                data: header_hash.as_bytes().to_vec(),
+            }))
+        }
+        StacksBlockInfoProperty::IdentityHeaderHash => {
+            let id_header_hash = env
+                .global_context
+                .database
+                .get_index_block_header_hash(height_value)?;
+            Value::Sequence(SequenceData::Buffer(BuffData {
+                data: id_header_hash.as_bytes().to_vec(),
+            }))
+        }
+    };
+
+    Value::some(result)
+}
+
+/// Handles the function `get-tenure-info?` special function.
+/// Interprets `args` as variables `[property-name, block-height]`, and returns
+/// a property value determined by `property-name`:
+/// - `time` returns the burn block time for the tenure of which `block-height` is a part
+/// - `vrf-seed` returns the VRF seed for the tenure of which `block-height` is a part
+/// - `burnchain-header-hash` returns header hash of the burnchain block corresponding to the tenure of which `block-height` is a part
+/// - `miner-address` returns the address of the principal that mined the tenure of which `block-height` is a part
+/// - `miner-spend-winner` returns the number of satoshis spent by the winning miner for the tenure of which `block-height` is a part
+/// - `miner-spend-total` returns the total number of satoshis spent by all miners for the tenure of which `block-height` is a part
+/// - `block-reward` returns the block reward for the tenure of which `block-height` is a part
+///
+/// # Errors:
+/// - CheckErrors::IncorrectArgumentCount if there aren't 2 arguments.
+/// - CheckErrors::GetTenureInfoExpectPropertyName if `args[0]` isn't a ClarityName.
+/// - CheckErrors::NoSuchTenureInfoProperty if `args[0]` isn't a TenureInfoProperty.
+/// - CheckErrors::TypeValueError if `args[1]` isn't a `uint`.
+pub fn special_get_tenure_info(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    // (get-tenure-info? property-name block-height-uint)
+    runtime_cost(ClarityCostFunction::BlockInfo, env, 0)?;
+
+    check_argument_count(2, args)?;
+
+    // Handle the block property name input arg.
+    let property_name = args[0]
+        .match_atom()
+        .ok_or(CheckErrors::GetTenureInfoExpectPropertyName)?;
+
+    let block_info_prop = TenureInfoProperty::lookup_by_name(property_name)
+        .ok_or(CheckErrors::GetTenureInfoExpectPropertyName)?;
+
+    // Handle the block-height input arg.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
+        Value::UInt(result) => Ok(result),
+        x => Err(CheckErrors::TypeValueError(TypeSignature::UIntType, x)),
+    }?;
+
+    let height_value = match u32::try_from(height_value) {
+        Ok(result) => result,
+        _ => return Ok(Value::none()),
+    };
```

Comment:
```suggestion
    let Ok(height_value) = u32::try_from(height_value) else {
        return Ok(Value::none())
    };
```

Reply:
Fixed in 1fad8d226e8081b4204c6a6ecc44c5ff8038013b